### PR TITLE
Don't save if nav items have empty labels

### LINKED ISSUE #4942: Navigation - Saving with empty label deletes item rather than erroring
## Steps to reproduce
1. Delete the label for one of your nav items (leaving the url)
2. Save

## Expected behaviour
1. An error message should pop up letting me know that I can't save with an empty label

### LINKED ISSUE #4944: Navigation - Saving with last field filled (but not added) makes ordering jump
This might be fixed by #4943, I haven't checked yet, just wanted to get it written in.

## Steps to reproduce
1. Fill in the last field (the one with no handle)
2. Save

## Expected
the new nav item stays at the bottom

## Observed
the nav item jumps halfway up the list of navitems

### PR DESCRIPTION
**Pull request by novaugust**
Closes #4942, Closes #4944

### CODE REVIEW

File: core/client/controllers/settings/navigation.js [hunk 1]
```diff
@@ -110,9 +111,21 @@ NavigationController = Ember.Controller.extend({
...
+            if (navItems.find(function (item) { return !item.get('isComplete') && !item.get('last');})) {
+                self.notifications.showErrors(['One of your navigation items has an empty label.<br>Please enter a new label or delete the item before saving.']);
+                return;
+            }
+            // The last item is typically ignored in ordering, give it
+            // the last spot in case it has a label & url and hasn't been
+            // added yet.
+            if (lastItem.get('order') != 0 && !lastItem.get('order')) {
+                lastItem.set('order', navItems.length);
+            }
```
Comment (novaugust) [inline core/client/controllers/settings/navigation.js]: This is the fix for #4944

File: core/client/controllers/settings/navigation.js [hunk 2]
```diff
@@ -110,9 +111,21 @@ NavigationController = Ember.Controller.extend({
...
+                navItems = this.get('navigationItems'),
+                lastItem = navItems.get('lastObject'),
                 match;
-
-            navSetting = this.get('navigationItems').map(function (item) {
+            // Don't save if there's a blank label.
+            if (navItems.find(function (item) { return !item.get('isComplete') && !item.get('last');})) {
+                self.notifications.showErrors(['One of your navigation items has an empty label.<br>Please enter a new label or delete the item before saving.']);
+                return;
+            }
```
Comment (novaugust) [inline core/client/controllers/settings/navigation.js]: Fixes #4942

File: core/client/controllers/settings/navigation.js [hunk 3]
```diff
@@ -110,9 +111,21 @@ NavigationController = Ember.Controller.extend({
...
+            // The last item is typically ignored in ordering, give it
+            // the last spot in case it has a label & url and hasn't been
+            // added yet.
+            if (lastItem.get('order') != 0 && !lastItem.get('order')) {
```
Comment (ErisDS) [inline core/client/controllers/settings/navigation.js]: `!=`is breaking the build ;)

**Comment by ErisDS** [comment]
This PR goes a bit too far with the validation I fear! It's not possible to make a home link:

**Comment by novaugust** [comment]
Harrumph. I'll look into this tomorrow

**Comment by novaugust** [comment]
I can has fixed!

**Pull request merged**